editor.planx.uk/src/components/Feedback.tsx

```diff
@@ -66,184 +67,330 @@ const FeedbackForm = styled("form")(({ theme }) => ({
 }));
 
 const FeedbackComponent: React.FC = () => {
-  return (
-    <>
+  enum FeedbackView {
+    PhaseBanner,
+    FeedbackTriage,
+    ReportAnIssue,
+    ShareAnIdea,
+    ShareAComment,
+    ThanksForFeedback,
+  }
+
+  enum FeedbackViewClickEvents {
+    Close,
+    Back,
+    OpenTriage,
+    OpenReportAnIssue,
+    OpenShareAnIdea,
+    OpenShareAComment,
+  }
+
+  const [currentFeedbackView, setCurrentFeedbackView] = useState<FeedbackView>(
+    FeedbackView.PhaseBanner,
+  );
+  const previousFeedbackView = usePrevious(currentFeedbackView);
+
+  function handleFeedbackViewClick(event: FeedbackViewClickEvents) {
```

Comment:
I tried implementing the handler as a single function with a switch statement which I think has worked pretty well?

Reply:
Agreed - nice and need and maintainable! Take a look at my comment below on how we could make this a little safer.